cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -10,23 +11,34 @@ import (
 )
 
 type CredentialsCmd struct {
-	Audience       string `required:"" help:"SVID JWT Audience. Must match AWS configuration"`
-	SpiffeID       string `required:"" help:"The SPIFFE ID of this workload"`
-	WorkloadSocket string `optional:"" help:"Path to SPIFFE Workload Socket"`
-	RoleARN        string `required:"" help:"AWS Role ARN to assume"`
-	SessionName    string `optional:"" help:"AWS Session Name"`
-	STSEndpoint    string `optional:"" help:"AWS STS Endpoint variable"`
+	Audience        string        `required help:"SVID JWT Audience. Must match AWS configuration"`
+	SpiffeID        string        `required help:"The SPIFFE ID of this workload"`
+	WorkloadSocket  string        `optional help:"Path to SPIFFE Workload Socket"`
+	RoleARN         string        `required help:"AWS Role ARN to assume"`
+	SessionName     string        `optional help:"AWS Session Name"`
+	STSEndpoint     string        `optional help:"AWS STS Endpoint variable"`
+	SessionDuration time.Duration `optional type:iso8601duration help:"AWS session duration in ISO8601 duration format (e.g. PT5M for five minutes)"`
 }
 
-func (c *CredentialsCmd) Run() error {
+type CliContext struct {
+	JWTSourceProvider credentials.JWTSourceProvider
+	STSProvider       credentials.STSProvider
+}
```

Comment:
Kong allows you to define a context structure and then pass it into the _Run()_ method which in turn passes the context as an argument to the _Run()_ method on your command type (_CLI_ in our case). This allows us to provide additional configuration orthogonally to what's determined simply by parsing the command line.